components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -151,14 +148,14 @@ func (g *gpsrtk) receiveAndWriteCorrectionData() {
 		return
 	}
 
-	g.reader, err = g.getStream()
+	reader, err := g.getStream()
 	if err != nil {
 		g.err.Set(err)
```

Comment:
We should probably look to clean up the LastError. I personally don't like the idea of returning an error that happened in another function from a different function. Besides it seems like we don't use it much. And when we are returning from a function we should probably return the error with the error message now that the GPS code is cleaner than before

Reply:
I'll make a ticket if you think it's a good idea moving forward

Reply:
Yes, I think it's a great idea! I've never liked the `LastError` because it mingles unrelated errors into unrelated functionality. I'm delighted to hear you think similarly!

Reply:
...oh wait, this time we set the error because we're in a background goroutine and can't return the error to whatever called us. I'll log the error instead.